builtin/providers/google/resource_compute_instance.go

```diff
@@ -673,23 +681,39 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 
 	// If the Metadata has changed, then update that.
 	if d.HasChange("metadata") {
-		metadata, err := resourceInstanceMetadata(d)
-		if err != nil {
-			return fmt.Errorf("Error updating metadata: %s", err)
-		}
-		op, err := config.clientCompute.Instances.SetMetadata(
-			config.Project, zone, d.Id(), metadata).Do()
-		if err != nil {
-			return fmt.Errorf("Error updating metadata: %s", err)
-		}
+		o, n := d.GetChange("metadata")
 
-		// 1 5 2
-		opErr := resourceOperationWaitZone(config, op, zone, "metadata to update")
-		if opErr != nil {
-			return opErr
+		updateMD := func() error {
+			// Reload the instance in the case of a fingerprint mismatch
+			instance, err = getInstance(config, d);
+			if err != nil {
+				return err
+			}
+
+			md := instance.Metadata
+
+			MetadataUpdate(o.(map[string]interface{}), n.(map[string]interface{}), md)
+
+			if err != nil {
+				return fmt.Errorf("Error updating metadata: %s", err)
+			}
+			op, err := config.clientCompute.Instances.SetMetadata(
+				config.Project, zone, d.Id(), md).Do()
+			if err != nil {
+				return fmt.Errorf("Error updating metadata: %s", err)
+			}
+
+			// 1 5 2
```

Comment:
I'm not sure, It was in the code before I made changes... It probably doesn't make sense for me to keep it though.